satpy/readers/viirs_edr_active_fires.py

```diff
@@ -43,11 +49,13 @@ def get_dataset(self, dsid, dsinfo):
             Dask DataArray: Data
 
         """
-        data = self[dsinfo.get('file_key', dsid.name)]
+
+        key = dsinfo.get('file_key', dsid.name).format(variable_prefix=self.prefix)
+        data = self[key]
         data.attrs.update(dsinfo)
 
-        data.attrs["platform_name"] = self['/attr/satellite_name']
-        data.attrs["sensor"] = self['/attr/instrument_name']
+        data.attrs["platform_name"] = self.get('/attr/satellite_name')
+        data.attrs["sensor"] = self.get('/attr/instrument_name')
```

Comment:
Does this ever not exist? These need to always be set to something not `None`.

Reply:
There was a case, should be resolved now